yt_dlp/extractor/callin.py

```diff
@@ -0,0 +1,135 @@
+            'ext': 'mp3',
+            'display_id': 'the-title-ix-regime-and-the-long-march-through-EBfXYSrsjc',
+            'thumbnail': 'https://d2xpqol0sc5yi.cloudfront.net/82cf84f4f7b70e3e1f9e36505cce56d97a31d3bd0589156595f362d25e012e06.png',
+            'description': 'First episode',
+            'uploader': 'Wesley Yang',
+            'timestamp': 1639404128.65,
+            'upload_date': '20211213',
+            'uploader_id': 'wesyang',
+            'uploader_url': 'http://wesleyyang.substack.com',
+            'channel': 'Conversations in Year Zero',
+            'channel_id': '436d1f82ddeb30cd2306ea9156044d8d2cfdc3f1f1552d245117a42173e78553',
+            'channel_url': 'https://callin.com/show/conversations-in-year-zero-oJNllRFSfx',
+            'duration': 9951.936,
+            # 'view_count': 1489,
+            'categories': ['News & Politics', 'History', 'Technology'],
+            'cast': ['Wesley Yang', 'KC Johnson', 'Gabi Brahm'],
+            'series': 'Conversations in Year Zero',
+            'series_id': '436d1f82ddeb30cd2306ea9156044d8d2cfdc3f1f1552d245117a42173e78553',
+            'episode': 'The Title IX Regime and the Long March Through and Beyond the Institutions',
+            'episode_number': 1,
+            'episode_id': '218b979630a35ead12c6fd096f2996c56c37e4d0dc1f6dc0feada32dcf7b31cd'
+        }
+    }]
+
+    def try_get_user_name(self, d):
+        names = [d.get(n) for n in ('first', 'last')]
+        if None in names:
+            return None
+        return ' '.join(names)
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        webpage = self._download_webpage(url, display_id)
+
+        json_string = self._html_search_regex(
+            '<script\\s+id\\s*=\\s*["\']__NEXT_DATA__["\'].*?>(.+?)</script>',
+            webpage, 'json')
+        j = self._parse_json(json_string, display_id)
+        episode = j['props']['pageProps']['episode']
+
+        id = episode['id']
+        title = episode.get('title')
+        url = episode['m3u8']
+        formats = self._extract_m3u8_formats(url, id, ext='mp3')
+        self._sort_formats(formats)
+
+        show = traverse_obj(episode, ('show', 'title'))
+        show_id = traverse_obj(episode, ('show', 'id'))
+
+        # get the show metadata for some supplementary info
+        sj = None
+        try:
+            app_slug = self._html_search_regex(
+                '<script\\s+src=["\']/_next/static/([a-zA-Z0-9_]+)/_',
+                webpage, 'app slug', fatal=False)
+            if app_slug is None:
+                # this sometimes works, but sometimes seems to lag changes in the actual urls used.
+                app_slug = j['buildId']
+            show_slug = episode['show']['linkObj']['resourceUrl'].rsplit('/', 1)[1]
+            show_json_url = "https://www.callin.com/_next/data/" + app_slug + "/show/" + show_slug + ".json"
```

Comment:
```suggestion
            show_json_url = f'https://www.callin.com/_next/data/{app_slug}/show/{show_slug}.json'
```
1. Single quotes everywhere
2. Avoid `+`ing strings

Reply:
Done (in 5a94e184a2cd538ea64ad9b6c242056692be706c)